### ISSUE TITLE
Checkboxs appear but are nonfunctional if markdown is malformed

### ISSUE THREAD

**Issue by saarantras** [laurent22/joplin#4070]
## Environment

Joplin version: Joplin 1.0.242 (prod, linux)
Platform: Desktop
OS specifics:  Linux Mint 18.3 Sylvia

## Steps to reproduce

1.  Paste the following into a joplin note (**preserving whitspace**)
```
- [ ] CORRECT

-  [ ] INCORRECT

```
2. Switch to dual view : so you can see markdown and the rendered markdown.

3. Click both checkboxes on the right-side rendered view

4. Examine the markdown : CORECT has an ```x``` character inserted in between the square brackets, while INCORRECT does not.

**Note the one space between the - and [ characters in "CORRECT", and the two spaces between the - and [ characters in "INCORRECT".**

## Describe what you expected to happen

I expected either "INCORRECT" to not render as a check box OR for it to render AND function properly. This is a very frustrating bug, because you can make a giant checklist, switch to the rendered view, check of items, and have them "randomly" uncheck.

## Logfile

Not necessary.

**Comment by a-ys** [comment]
This occurs for all checkboxes with 2-4 spaces between the hyphen and the first square bracket. I.e, the following all cause the same error: 
```
-  [ ] CORRECT
-   [ ] CORRECT
-    [ ] CORRECT
```
After this, the markdown-it parser identifies it as an indented code block, although it is not displayed as such in the renderer.

@laurent22 I would be happy to fix this bug. What is the preferred behavior here?

**Comment by laurent22** [comment]
Yes feel free to create a pull request for this and please add a test unit. Only `- [ ]` (one space) should be supported.

**Comment by laurent22** [comment]
I'm re-opening this and tagging as "backlog" so that it doesn't get auto-closed by stale bot. Would still accept a PR if someone can come up with a simple fix.
